File: JAVA/Odd Even Problem.java
class Solution {
    public static String oddEven(String s) {
        int charArr[]=new int[27];
        for(int i =0;i<s.length();i++){
            char ch=s.charAt(i);
            charArr[ch-'a']++;
        }
        int x=0;
        int y=0;
        
        for (int i=0; i<27;i++){
            if(charArr[i]!=0 && charArr[i]%2==0 && (i+1)%2==0)x++;
            else if(charArr[i]%2!=0 && (i+1)%2!=0)y++;
    
        }
        int ans= x+y;
        return (ans&1)==1?"ODD":"EVEN";
        
    }
}

File: C++/Odd Even Problem.cpp
class Solution {
  public:
    string oddEven(string s) {
        vector<int> v(27, 0);
        int n = s.size();
        int count = 0;
        for(int i=0; i<n; i++){
            v[1+s[i]-'a']++;
        }
        for(int i=1; i<27; i++){
            if(v[i] > 0){
                if(i%2 == 0){
                    if(v[i]%2 == 0){
                        count++;
                    }
                }
                else{
                    if(v[i]%2 != 0){
                        count++;
                    }
                }
            }
        }
        if(count%2 == 0)
            return "EVEN";
        return "ODD";
    }
};
